pageserver/src/layered_repository.rs

```diff
@@ -1237,6 +1240,7 @@ impl LayeredTimeline {
         drop(layers);
         drop(write_guard);
 
+        //FIXME peek_oldest_open() above is global, so we need to sync not only our timelinedir.
```

Comment:
@hlinnaka,
IIUC, `peek_oldest_open()` is global now and so is checkpointer.
So `timeline_dir.sync_all()` is not correct anymore,  we need to fsync checkpointed files one by one.

To fix it globally, we can either detach checkpointer from tenant (all checkpointers serve all tenants) or add a filter to `peek_oldest_open()`. What do you think is better?

Reply:
> IIUC, peek_oldest_open() is global now and so is checkpointer.

No, the peek_oldest_open() is still per-timeline. There's a separate LayerMap, and a binary heap in each LayerMap, for each timeline.

All the open layers are kept in the global array, though. But in fact that's not really necessary anymore with this PR. The global layer map is kind of obsolete with this PR, actually. We don't really make any decisions on the layers globally anymore

Reply:
> All the open layers are kept in the global array, though. But in fact that's not really necessary anymore with this PR. The global layer map is kind of obsolete with this PR, actually. We don't really make any decisions on the layers globally anymore

Thank you for the explanation. 
Is there any use in keeping global map for the future or I can remove it as a part of the PR cleanup?